extensions/wikia/UserProfilePageV3/js/UserProfilePage.js

```diff
@@ -414,248 +401,263 @@ var UserProfilePage = {
 				controller: 'UserProfilePage',
-				method: 'removeavatar',
-				format: 'json',
+				method: 'getClosingModal',
+				type: 'GET',
 				data: {
-					avUser: name,
-					token: mw.user.tokens.get('editToken')
-				},
-				callback: function (data) {
+					userId: UserProfilePage.userId,
+					rand: Math.floor(Math.random() * 100001)
+				}
+			}).then(function (data) {
+				var id = $(data.body).attr('id') + 'Wrapper',
+					modalConfig = {
+						vars: {
+							id: id,
+							content: data.body,
+							size: 'medium',
+							title: mw.message('userprofilepage-closing-popup-header').escaped(),
+							buttons: [{
+								vars: {
+									value: mw.message('userprofilepage-closing-popup-save-and-quit').escaped(),
+									classes: ['normal', 'primary'],
+									data: [{
+										key: 'event',
+										value: 'save'
+									}]
+								}
+							}, {
+								vars: {
+									value: mw.message('userprofilepage-closing-popup-discard-and-quit').escaped(),
+									data: [{
+										key: 'event',
+										value: 'quit'
+									}]
+								}
+							}, {
+								vars: {
+									value: mw.message('userprofilepage-closing-popup-cancel').escaped(),
+									data: [{
+										key: 'event',
+										value: 'close'
+									}]
+								}
+							}]
+						}
+					};
+
+				UserProfilePage.modalComponent.createComponent(modalConfig, function (confirmationModal) {
+
+					// attaching handlers to button events
+					confirmationModal.bind('save', function () {
+						confirmationModal.trigger('close');
+						UserProfilePage.saveUserData();
+					});
+					confirmationModal.bind('quit', function () {
+						UserProfilePage.wasDataChanged = false;
+						UserProfilePage.modal.trigger('close');
+						confirmationModal.trigger('close');
+					});
+
+					confirmationModal.show();
+
+				});
+			});
+		},
+
+		removeAvatar: function (name, question) {
+			var answer = window.confirm(question);
+
+			if (answer) {
+				nirvana.sendRequest({
+					controller: 'UserProfilePage',
+					method: 'removeavatar',
+					type: 'POST',
+					data: {
+						avUser: name,
+						token: mw.user.tokens.get('editToken')
+					}
+				}).then(function (data) {
 					if (data.status === 'ok') {
 						window.location.reload();
 					} else {
 						UserProfilePage.error(data.error);
 					}
-				}
-			});
+				});
+			}
+		},
+
+		/**
+		 * Clear all contents from user profile (including user avatar)
+		 * @param {string} targetUser Name of target user
+		 * @param {string} confirmationMessage Confirmation message to display
+		 */
+		clearMastheadContents: function (targetUser, confirmationMessage) {
+			var confirmClear = window.confirm(confirmationMessage)
+			if (confirmClear) {
+				nirvana.sendRequest({
+					controller: 'UserProfilePage',
+					method: 'clearMastheadContents',
+					type: 'POST',
+					data: {
+						target: targetUser,
+						token: mw.user.tokens.get('editToken')
+					}
+				}).done(function (res) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameter "res". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)